packages/aws-rfdk/lib/deadline/lib/sep-configuration.ts

```diff
@@ -0,0 +1,398 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path';
+
+import {
+  IVpc,
+  SubnetSelection,
+  SubnetType,
+} from '@aws-cdk/aws-ec2';
+import {
+  Code,
+  Function as LambdaFunction,
+  LayerVersion,
+  Runtime,
+} from '@aws-cdk/aws-lambda';
+import {
+  RetentionDays,
+} from '@aws-cdk/aws-logs';
+import {
+  ISecret,
+} from '@aws-cdk/aws-secretsmanager';
+import {
+  Construct,
+  CustomResource,
+  Duration,
+  Stack,
+} from '@aws-cdk/core';
+import {
+  ARNS,
+} from '../../lambdas/lambdaLayerVersionArns';
+import {
+  ISEPConfiguratorResourceProperties,
+} from '../../lambdas/nodejs/sep-configuration';
+import {
+  IRenderQueue,
+} from './render-queue';
+import {
+  SEPSpotFleet,
+} from './sep-spotfleet';
+
+/**
+ * How the event plug-in should respond to events.
+ */
+export enum SpotEventPluginState {
+  /**
+   * All jobs and Workers will trigger the events for this plugin.
+   */
+  GLOBAL_ENABLED = 'Global Enabled',
+
+  /**
+   * No events are triggered for the plugin.
+   */
+  DISABLED = 'Disabled',
+}
+
+/**
+ * Different logging levels.
+ */
+export enum SpotEventPluginLoggingLevel {
+  /**
+   * Standard logging level.
+   */
+  STANDARD = 'Standard',
+
+  /**
+   * Detailed logging about the inner workings of the Spot Event Plugin.
+   */
+  VERBOSE = 'Verbose',
+
+  /**
+   * All Verbose logs plus additional information on AWS API calls that are used.
+   */
+  DEBUG = 'Debug',
+
+  /**
+   * No logging enabled.
+   */
+  OFF = 'Off',
+}
+
+/**
+ * How the Spot Event Plugin should handle Pre Job Tasks.
+ */
+export enum SpotEventPluginPreJobTaskMode {
+  /**
+   * Only start 1 Spot instance for the pre job task and ignore any other tasks for that job until the pre job task is completed.
+   */
+  CONSERVATIVE = 'Conservative',
+
+  /**
+   * Do not take the pre job task into account when calculating target capacity.
+   */
+  IGNORE = 'Ignore',
+
+  /**
+   * Treat the pre job task like a regular job queued task.
+   */
+  NORMAL = 'Normal',
+}
+
+/**
+ * The Worker Extra Info column to be used to display AWS Instance Status
+ * if the instance has been marked to be stopped or terminated by EC2 or Spot Event Plugin.
+ * See "AWS Instance Status" option at https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/event-spot.html?highlight=spot%20even%20plugin#event-plugin-configuration-options
+ */
+export enum SpotEventPluginAwsInstanceStatus {
+  DISABLED = 'Disabled',
+  EXTRA_INOF_0 = 'ExtraInfo0',
+  EXTRA_INOF_1 = 'ExtraInfo0',
+  EXTRA_INOF_2 = 'ExtraInfo0',
+  EXTRA_INOF_3 = 'ExtraInfo0',
+  EXTRA_INOF_4 = 'ExtraInfo0',
+  EXTRA_INOF_5 = 'ExtraInfo0',
+  EXTRA_INOF_6 = 'ExtraInfo0',
+  EXTRA_INOF_7 = 'ExtraInfo0',
+  EXTRA_INOF_8 = 'ExtraInfo0',
+  EXTRA_INOF_9 = 'ExtraInfo0',
+}
+
+/**
+ * Spot Event Plugin configuration options
```

Comment:
All the default values I used here are actually set by Deadline only during the first launch, when there are no such entries in the database. But if any property was previously set, and then on the next deployment we don't set it, the previous value will be used, as we don't `reset` the values.

For example:
Deploy1: `enableResourceTracker` not provided, so Deadline will use `true` by default.
Deploy 2: set `enableResourceTracker` to `false`. This will update configurations.
Deploy 3: `enableResourceTracker` not provided, so Deadline will not change the previously set `false` value.

Should we add the description of this behaviour here? Otherwise customers might expect, that if the property is not provided, it will be always set to default value.

Reply:
Yes, I fully agree. Enforced all the defaults.
Before I just I confused myself that this might surprise Deadline customers, but this Deadline's behavior is actually hidden from them.

Reply:
Resolved. Confirmed that the defaults are being applied as documented